### PR TITLE
[ISSUE-869][REFACTOR] Migrate partition size/sorter related conf to Celeborn ConfigEntity

### LINKED ISSUE #869: [REFACTOR] Migrate partition size/sorter related conf to Celeborn ConfigEntity
### Describe the intention you'd like
A clear and concise description of what you want to do.

/cc @who-need-to-know

/assign @who-can-help-you

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### Related issues.
close #869

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
gentle ping @pan3793 @waitinfuture

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -1463,21 +1487,42 @@ object CelebornConf extends Logging {
...
-  }
 
-  def minimumPartitionSizeForEstimation(conf: CelebornConf): Long = {
-    Utils.byteStringAsBytes(conf.get("rss.minimum.estimate.partition.size", "8m"))
-  }
+  val INITIAL_ESTIMATED_PARTITION_SIZE: ConfigEntry[Long] =
+    buildConf("celeborn.worker.initialEstimatedPartitionSize")
+      .withAlternative("rss.initial.partition.size")
+      .categories("master")
+      .doc("Initial estimated partition size, default is 64m, and it will change according to runtime stats.")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1495]: remove "default is 64m"

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -1463,21 +1487,42 @@ object CelebornConf extends Logging {
...
+      .version("0.2.0")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefaultString("64mb")
 
-  def partitionSizeUpdaterInitialDelay(conf: CelebornConf): Long = {
-    Utils.timeStringAsMs(conf.get("rss.partition.size.update.initial.delay", "5m"))
-  }
+  val MINIMUM_ESTIMATED_PARTITION_SIZE: ConfigEntry[Long] =
+    buildConf("celeborn.worker.minimumEstimatedPartitionSize")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1501]:
```suggestion
    buildConf("celeborn.worker.minEstimatedPartitionSize")
```

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -1463,21 +1487,42 @@ object CelebornConf extends Logging {
...
-    Utils.timeStringAsMs(conf.get("rss.partition.size.update.interval", "10m"))
-  }
+  val ESTIMATED_PARTITION_SIZE_UPDATE_INITIAL_DELAY: ConfigEntry[Long] =
+    buildConf("celeborn.worker.estimatedPartitionSize.update.initialDelay")
+      .withAlternative("rss.partition.size.update.initial.delay")
+      .categories("master")
+      .doc("Initial delay time before start updating estimated partition size.")
+      .version("0.2.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("5m")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1516]: better to use 5min, to distinguish w/ megabytes

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 4]
```diff
@@ -1463,21 +1487,42 @@ object CelebornConf extends Logging {
...
+
+  val ESTIMATED_PARTITION_SIZE_UPDATE_INTERVAL: ConfigEntry[Long] =
+    buildConf("celeborn.worker.estimatedPartitionSize.update.interval")
+      .withAlternative("rss.partition.size.update.interval")
+      .categories("master")
+      .doc("Interval of updating estimated partition size.")
+      .version("0.2.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("10m")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1525]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 5]
```diff
@@ -1286,6 +1292,33 @@ object CelebornConf extends Logging {
...
+    buildConf("celeborn.worker.partitionSorter.sort.timeout")
+      .withAlternative("rss.partition.sort.timeout")
+      .categories("worker")
+      .doc("Timeout for a shuffle file to sort.")
+      .version("0.2.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("220s")
+
+  val INIT_SINGLE_PARTITION_SORTER_RESERVE_MEMORY: ConfigEntry[Long] =
+    buildConf("celeborn.worker.partitionSorter.initialReserveSingleSortMemory")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1314]:
```suggestion
    buildConf("celeborn.worker.partitionSorter.initialReservedMemoryPerPartition")
```

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1495]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 3 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1516]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 5 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1314]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 6]
```diff
@@ -1463,21 +1488,41 @@ object CelebornConf extends Logging {
       .checkValue(Seq("roundrobin", "loadaware").contains(_), "")
       .createWithDefault("roundrobin")
 
-  def initialPartitionSize(conf: CelebornConf): Long = {
-    Utils.byteStringAsBytes(conf.get("rss.initial.partition.size", "64m"))
-  }
+  val INITIAL_ESTIMATED_PARTITION_SIZE: ConfigEntry[Long] =
+    buildConf("celeborn.worker.initialEstimatedPartitionSize")
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1492]: This config is not intended for the worker. May be the "celeborn.master" will be a better namespace?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 7]
```diff
@@ -1463,21 +1488,41 @@ object CelebornConf extends Logging {
...
+      .doc("Ignore partition size smaller than this configuration for partition size estimation.")
+      .version("0.2.0")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefaultString("8mb")
 
-  def partitionSizeUpdaterInitialDelay(conf: CelebornConf): Long = {
-    Utils.timeStringAsMs(conf.get("rss.partition.size.update.initial.delay", "5m"))
-  }
+  val ESTIMATED_PARTITION_SIZE_UPDATE_INITIAL_DELAY: ConfigEntry[Long] =
+    buildConf("celeborn.worker.estimatedPartitionSize.update.initialDelay")
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1510]: I think this config is for master.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 8]
```diff
@@ -1463,21 +1488,41 @@ object CelebornConf extends Logging {
...
+      .doc("Initial delay time before start updating estimated partition size.")
+      .version("0.2.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("5min")
 
-  def partitionSizeUpdateInterval(conf: CelebornConf): Long = {
-    Utils.timeStringAsMs(conf.get("rss.partition.size.update.interval", "10m"))
-  }
+  val ESTIMATED_PARTITION_SIZE_UPDATE_INTERVAL: ConfigEntry[Long] =
+    buildConf("celeborn.worker.estimatedPartitionSize.update.interval")
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1519]: ditto.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 9]
```diff
@@ -1463,21 +1488,41 @@ object CelebornConf extends Logging {
...
+      .doc("Initial estimated partition size, it will change according to runtime stats.")
+      .version("0.2.0")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefaultString("64mb")
 
-  def minimumPartitionSizeForEstimation(conf: CelebornConf): Long = {
-    Utils.byteStringAsBytes(conf.get("rss.minimum.estimate.partition.size", "8m"))
-  }
+  val MINIMUM_ESTIMATED_PARTITION_SIZE: ConfigEntry[Long] =
+    buildConf("celeborn.worker.minEstimatedPartitionSize")
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1501]: How about "celeborn.worker.minPartitionSizeForEstimation"? I think this name can match the doc.

Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1501]: MinEstimatedPartitionSize looks like that this value is the min of estimated partition size.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 6 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1492]: Forgot change. ==

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 9 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Done

**Comment by AngersZhuuuu** [comment]
how about current? @FMX @pan3793

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 10]
```diff
@@ -2027,21 +2071,6 @@ object CelebornConf extends Logging {
       .timeConf(TimeUnit.SECONDS)
       .createWithDefaultString("10s")
 
-  def initialReserveSingleSortMemory(conf: CelebornConf): Long = {
-    conf.getSizeAsBytes("rss.worker.initialReserveSingleSortMemory", "1mb")
-  }
-
-  def defaultStorageType(conf: CelebornConf): StorageInfo.Type = {
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2034]: Remove unused

**Comment by AngersZhuuuu** [comment]
Ping again

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java [hunk 11]
```diff
@@ -101,9 +101,9 @@ public class PartitionFilesSorter extends ShuffleRecoverHelper {
 
   public PartitionFilesSorter(
       MemoryTracker memoryTracker, CelebornConf conf, AbstractSource source) {
-    this.sortTimeout = CelebornConf.partitionSortTimeout(conf);
+    this.sortTimeout = conf.partitionSorterSortPartitionTimeout();
     this.shuffleChunkSize = conf.shuffleChunkSize();
-    this.initialReserveSingleSortMemory = CelebornConf.initialReserveSingleSortMemory(conf);
+    this.reserveMemoryPerPartition = conf.partitionSorterReserveMemoryPerPartition();
```
Comment (pan3793) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java:106]: reserve => reserved

**Review by pan3793 — APPROVED** [review]
LGTM except to minor one

**Review by FMX — APPROVED** [review]
LGTM.

**Pull request merged**